### PR TITLE
Implementing serializer

### PR DESCRIPTION
**Pull request by Gmarn**
# Summary

- Adding gem 'active_model_serializer';
- Managing new routes for books in appi;
- Dealing with serializer for books;

# Trello card

https://trello.com/c/vk1awnxu

# Remarks

Warning: I did not success to have the exact same output for api/v1/book/id

### CODE REVIEW

File: app/controllers/api/v1/books_controller.rb [hunk 1]
```diff
@@ -0,0 +1,15 @@
+module Api
+  module V1
+    class BooksController < ApiController
+      def index
+        @book = Book.all
```
Comment (alanhala) [inline app/controllers/api/v1/books_controller.rb]: You are returing a list of books, so let's call the variable `@books`

File: app/serializers/api/v1/rent_serializer.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+module Api
+  module V1
+    class RentSerializer < ActiveModel::Serializer
+      attributes :from, :to, :user
```
Comment (alanhala) [inline app/serializers/api/v1/rent_serializer.rb]: no :id for this serializer?

File: app/serializers/api/v1/} [hunk 3]
```diff
@@ -0,0 +1,10 @@
+module Api
```
Comment (alanhala) [inline app/serializers/api/v1/}]: Remove this file

File: app/serializers/api/v1/user_serializer.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+module Api
+  module V1
+    class UserSerializer < ActiveModel::Serializer
+      attributes :id, :first_name, :last_name, :email
```
Comment (alanhala) [inline app/serializers/api/v1/user_serializer.rb]: I've just checked in apiary and the User model has an image_url, rents_counter, comments_counter and unread_notifications_count. Shouldn't we add those attributes to the User?

Comment (alanhala) [inline app/serializers/api/v1/user_serializer.rb]: We must check if those attributes are added later in the training

Comment (Gmarn) [inline app/serializers/api/v1/user_serializer.rb]: You seem to be right, I do not see any card after that do this update. So i am going to create this queries.

Comment (Gmarn) [inline app/serializers/api/v1/user_serializer.rb]: There are not added later in the training, the notification table are not too. So I do not what doing?

Comment (Gmarn) [inline app/serializers/api/v1/user_serializer.rb]: And i do not think so...

Comment (alanhala) [inline app/serializers/api/v1/user_serializer.rb]: Ok, let's add those fields, so that the rendered json is the same as the one in apiary.

Comment (Gmarn) [inline app/serializers/api/v1/user_serializer.rb]: So I had it with a migration on the db or only in the serializer ?

Comment (Gmarn) [inline app/serializers/api/v1/user_serializer.rb]: add*

File: app/models/book.rb [hunk 5]
```diff
@@ -1,5 +1,6 @@
 class Book < ApplicationRecord
   has_many :rent
```
Comment (Dinuuu) [inline app/models/book.rb]: I think you should erase this line

File: app/models/rent.rb [hunk 6]
```diff
@@ -1,5 +1,5 @@
 class Rent < ApplicationRecord
-  belongs_to :user
+  belongs_to :user, :counter_cache => true
```
Comment (Dinuuu) [inline app/models/rent.rb]: use the new syntax for hashes `key: value` instead of `key => value`

File: app/serializers/book_serializer.rb [hunk 7]
```diff
@@ -0,0 +1,8 @@
+    class BookSerializer < ActiveModel::Serializer
```
Comment (Dinuuu) [inline app/serializers/book_serializer.rb]: Check indentation in this file

File: app/serializers/index_book_serializer.rb [hunk 8]
```diff
@@ -0,0 +1,4 @@
+class IndexBookSerializer < ActiveModel::Serializer
```
Comment (Dinuuu) [inline app/serializers/index_book_serializer.rb]: I would name this `SimpleBookSerializer` and extend the `BookSerializer` from this

File: app/serializers/rent_serializer.rb [hunk 9]
```diff
@@ -0,0 +1,4 @@
+class RentSerializer < ActiveModel::Serializer
+  attributes :id, :from, :to#, :user
```
Comment (Dinuuu) [inline app/serializers/rent_serializer.rb]: don't commit commented code, just erase it

File: app/models/book.rb (hunk 5 above)
Comment (Gmarn) [inline app/models/book.rb]: Without that not work anymore :/

File: app/serializers/simple_book_serializer.rb [hunk 10]
```diff
@@ -0,0 +1,8 @@
+class SimpleBookSerializer < ActiveModel::Serializer
```
Comment (Dinuuu) [inline app/serializers/simple_book_serializer.rb]: The change I meant was other. It was to rename `IndexBookSerializer` to `SimpleBookSerializer` and extend the `BookSerializer` from `SimpleBookSerializer` and delete the attributes that are no longer needed

**Pull request merged**